### PR TITLE
Add `impl From<()> for Value`

### PR DESCRIPTION
**Pull request by Nilix007**
Hey there,

this tiny PR adds the conversion from `()` to `Value::Null`.

I feel this is justified as serde_json already perceives `()` as the JSON value `null`:
```
println!("{:?}", serde_json::from_str::<()>("null"));
// prints `Ok(())`
    
println!("{:?}", serde_json::to_string(&()));
// prints `Ok("null")`
```
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0eac935756f3537acdf45486dcabb4db

Cheers,
Felix

PS: Please have a look at the marvelous signature of `<Value as From<()>>::from` :laughing:

### CODE REVIEW

**Comment by Nilix007** [comment]
Mh. So I was working on a library which uses diesel's [`Jsonb`](https://docs.rs/diesel/1.4.3/diesel/pg/types/sql_types/struct.Jsonb.html) internally for some kind of dynamic data field.  As `Jsonb` maps directly to `Value`, I'm exporting functions like `fn do_something<T: Into<Value>>(additional_data: T) {}` so that I can avoid pulling in the complete serde dependency tree and skip the serialization step completely (yay, one error case less on my side due to type safety).

Anyway, long story short, I think that the conversion from this PR would be quite convenient for the end user of my lib. And as it kinda feels "natural" to me, I sent this PR.

**Comment by dtolnay** [comment]
What do you mean by "pulling in the complete serde dependency tree"? If you depend on serde_json to use Value, that depends on serde anyway.

Do you prefer Into\<Value\> over:

```rust
fn do_something(additional_data: impl Serialize)
```

which would be more flexible?

**Comment by Nilix007** [comment]
> What do you mean by "pulling in the complete serde dependency tree"? If you depend on serde_json to use Value, that depends on serde anyway.
> 
Err, you're right, of course.  Please ignore this part :grimacing: 

> Do you prefer Into<Value> over:
> 
> ```rust
> fn do_something(additional_data: impl Serialize)
> ```
> 
> which would be more flexible?

Yes, because serde's serialization (and `serde_json::to_value` respectively) can fail while From/Into cannot.

**Comment by dtolnay** [comment]
Published in 1.0.42.
